pennylane/qnode.py

```diff
@@ -906,7 +905,9 @@ def construct(self, args, kwargs):  # pylint: disable=too-many-branches
         if any(isinstance(m, MidMeasureMP) for m in self.tape.operations):
             self._tape = qml.defer_measurements(self._tape)
 
-        if self.expansion_strategy == "device":
+        if self.expansion_strategy == "device" and not isinstance(
+            self.device, qml.devices.experimental.Device
+        ):
             self._tape = self.device.expand_fn(self.tape, max_expansion=self.max_expansion)

```

Comment:
```suggestion
        if self.expansion_strategy == "device":
            if isinstance(self.device, qml.Device):
                self._tape = self.device.expand_fn(self.tape, max_expansion=self.max_expansion)
           else:
              tapes, _, _ = self.device.preprocess(self.tape)
              if len(tapes) > 1:
                  raise ValueError("expansion_strategy=device is not supported when device preprocessing includes a batch transform")
              self._tape = tapes[0]
```

Maybe something like that?

Reply:
I had this before but I think it broke something, and it.. isn't really necessary. We're going to apply `preprocess` anyway so I'm fine to just skip it. Also I chatted with Jay about `max_expansion`, and he agreed with what I mentioned in standup (that the important thing is really a target gateset)